package.json

```diff
@@ -10,7 +10,9 @@
     "prepack": "node ./scripts/prepack.js",
     "clean": "rimraf ./coverage",
     "lint": "bash ./scripts/commitlint.sh && eslint --ignore-path .gitignore .",
+    "pretest": "node ./scripts/pretest.js",
```

Comment:
This unfortunately means that you /have/ to run the entire test suite with `npm test`, this prevents using mocha directly or testing single files such as with editor test integrations. 

Did you consider perhaps using mocha's `--require` option (inside `test/mocha.opts`) to require this file instead? That way it is ran from the test runner instead of `npm` during mocha load.

![screenshot 2019-02-20 at 09 56 04](https://user-images.githubusercontent.com/44164/53083167-aa98f800-34fe-11e9-9446-213889c1bea1.png)

(Side note, `npm test` is too slow as it is.. https://twitter.com/searls/status/1097275240030392321). Alternatively, `fixture` method could lazy parse and cache the result, that way when I am running a subset of tests (or unit tests that don't need fixtures at all) no wasted time happens of waiting for parsing. Perhaps to go further, you could keep the parse fixtures in memory instead of writing them to disk and wasting SSD write cycles?

Reply:
> Did you consider perhaps using mocha's `--require` option

Good idea! I could try that.

> `fixture` method could cache the result

I had a primitive cache in place at least for fixtures used at multiple places, such as `ordinary`, but I bumped into the "cannot freeze frozen API Elements" limitation.

> `fixture` method could lazy parse the result

Wouldn't that make `fixture` asynchronous again, which is exactly what I wanted to get rid of?

-----

I didn't go much into optimizing the result, as I believe the benefit of readability of the tests is higher than the lost second, especially for such a small test suite (it's not minutes, it's one second).

I think the ultimate solution is to parse the fixtures only when they change, i.e. "on write", not "on read". That's hard to achieve though, with the package workflow.

A primitive, annoying way would be to only parse the fixtures on CI, so it's up to the developer to re-parse the fixtures on their change, and if they fail to do so and the tests fail, they'll know at least at the time they issue a Pull Request. 

A more sophisticated method could be to "checksum" the fixture files, note the value to a throwaway gitignored file (`fixtures/.checksum`), and then proceed to re-parsing them only if it doesn't match (= fixture files had changed). In that way one would experience the extra second only after changing fixtures or after cloning the repo, i.e. also always on CI.

I'm still not sure these mechanisms are worth my time at this moment, but I'm interested in what you think about them. We could choose the one we both like and then file it as an issue here, documenting a possible improvement.